app/src/lib/app-state.ts

```diff
@@ -755,6 +755,13 @@ export interface ICherryPickState {
    * cherry pick.
    */
   readonly userHasResolvedConflicts: boolean
+
+  /**
+   * The sha of the target branch tip before cherry pick initiated.
+   *
+   * This will be set to null if no cherry pick has been initiated.
```

Comment:
If I'm reading this PR correctly, this only works if the cherry-pick was initiated from GitHub Desktop, right?

So maybe we should specify that here, if there is no way to get the original sha in that scenario.
```suggestion
   * This will be set to null if no cherry pick has been initiated, or if it was
   * initiated from outside of Desktop.
```

Reply:
Very good point! Thank you! And fortunately, that sha is stored in the sequencer during an active commit. I just added it to the setting progress based on state method.

Reply:
Yay!! 🎉